core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -337,6 +338,40 @@ public KeyStore getPrimaryKeyStore() throws Exception {
             throw new CarbonException("Permission denied for accessing primary key store");
         }
     }
+
+    /**
+     * Load the internal key store, this is allowed only for the super tenant
+     *
+     * @return internal key store object
+     * @throws Exception Carbon Exception when trying to call this method from a tenant other
+     *                   than tenant 0
+     */
+    public KeyStore getInternalKeyStore() throws Exception {
+
+        if (tenantId == MultitenantConstants.SUPER_TENANT_ID) {
+            if (internalKeyStore == null) {
+                ServerConfigurationService config = this.getServerConfigService();
+                if (config.
+                        getFirstProperty(RegistryResources.SecurityManagement.SERVER_INTERNAL_KEYSTORE_FILE) == null) {
+                    return null;
+                }
+                String file = new File(config
+                        .getFirstProperty(RegistryResources.SecurityManagement.SERVER_INTERNAL_KEYSTORE_FILE))
+                        .getAbsolutePath();
+                KeyStore store = KeyStore.getInstance(config
+                        .getFirstProperty(RegistryResources.SecurityManagement.SERVER_INTERNAL_KEYSTORE_TYPE));
+                String password = config
+                        .getFirstProperty(RegistryResources.SecurityManagement.SERVER_INTERNAL_KEYSTORE_PASSWORD);
+                try (FileInputStream in = new FileInputStream(file)){
+                    store.load(in, password.toCharArray());
+                    internalKeyStore = store;
+                }
+            }
+            return internalKeyStore;
+        } else {
+            throw new CarbonException("Permission denied for accessing internal key store");
```

Comment:
Is this the correct message to provide here. The internal key store is available only for the super tenant. Shouldn't that information should be mentioned.

Reply:
Changed the error message. Changed it to already existing codes as well.